src/modules/31-filestore/common/useDelete/useDelete.tsx

```diff
@@ -51,29 +54,48 @@ const useDelete = (identifier: string, name: string, type: string, notCurrentNod
     queryParams
   })
 
-  const { openDialog: openReferenceErrorDialog } = useConfirmationDialog({
-    contentText: (
-      <span>
-        {errorMessage.replace(/\[.+\]/g, name)}
-        {getString('filestore.fileReferenceText')}
-      </span>
-    ),
-    titleText: getString('filestore.cantDeleteFile'),
-    confirmButtonText: type === FileStoreNodeTypes.FILE ? getString('common.referenceButtonText') : undefined,
-    cancelButtonText: getString('cancel'),
-    intent: Intent.DANGER,
-    onCloseDialog: async (isConfirmed: boolean) => {
-      if (isConfirmed) {
-        setErrorMessage('')
-        setCurrentNode({
-          identifier,
-          name,
-          type: FileStoreNodeTypes.FILE,
-          children: []
-        } as FileStoreNodeDTO)
-        setActiveTab(FILE_VIEW_TAB.REFERENCED_BY)
+  const onDelete = async (forceDelete: boolean) => {
+    try {
+      const deleted = await deleteFile(identifier || '', {
+        headers: { 'content-type': 'application/json' },
+        queryParams: {
+          ...queryParams,
+          forceDelete: forceDelete ? forceDeleteEnabled : undefined
+        }
+      })
+      if (deleted) {
+        addDeletedNode(identifier)
+        showSuccess(getString('filestore.deletedSuccessMessage', { name: name, type: _capitalize(type) }))
+        getNode(nodeParams as FileStoreNodeDTO)
       }
+    } catch (err) {
+      handleFileDeleteError(err?.data.code, _defaultTo(err?.data?.message, err?.message))
```

Comment:
I'm never a big fan of using a 3rd party dependency to do trivially simple things that JS supports.
```suggestion
      handleFileDeleteError(err?.data.code, err?.data?.message || err?.message)
```